src/shared/dropdown-list-v2/nested-dropdown-list-helpers.ts

```diff
@@ -198,16 +192,14 @@ export const toggleSubtree = <T>(
 
 export const updateSelectedState = <T>(
     list: NestedDropdownListLocalItem<T>[],
-    selectedKeyPaths: string[][],
+    selectedKeyPaths: Set<string>,
     multiSelect: boolean
 ) => {
-    return produce(list, (draft) => {
+    const res = produce(list, (draft) => {
         for (let i = draft.length - 1; i >= 0; i--) {
             const item = draft[i];
-            item.checked = !!selectedKeyPaths.find((keyPath) =>
-                isEqual(keyPath, item.keyPath)
-            );
-
+            const itemKeyPathString = item.keyPath.join(',');
```

Comment:
can have a common helper `buildKeyPath` that takes in the keyPath array and performs the join, mainly to keep the separator consistent